packages/esm-appointments-app/src/appointments/scheduled/scheduled-appointments.component.tsx

```diff
@@ -93,19 +93,27 @@ const ScheduledAppointments: React.FC<ScheduledAppointmentsProps> = ({ visits, a
 };
 
 function useAllowedExtensions() {
-  const [allowedExtensions, setAllowedExtensions] = useState<Readonly<Record<string, boolean>>>({});
+  const [allowedExtensions, dispatch] = useReducer(
+    (state: Record<string, boolean>, action: { type: 'show_extension' | 'hide_extension'; extension: string }) => {
+      let addedState = {} as Record<string, boolean>;
+      switch (action.type) {
+        case 'show_extension':
+          addedState[action.extension] = true;
+          break;
+        case 'hide_extension':
+          addedState[action.extension] = false;
+          break;
+      }
 
-  const set = useCallback(
-    (value: boolean, key: string) => {
-      setAllowedExtensions({ ...allowedExtensions, [key]: value });
+      return { ...state, ...addedState };
     },
-    [allowedExtensions, setAllowedExtensions],
+    {},
   );
 
   return {
-    allowedExtensions,
-    showExtension: set.bind(null, true) as (extension: string) => void,
```

Comment:
I actually think the `.bind()` here on a function returned from `useCallback()` was probably the issue. However, `useReducer()` exists exactly for these cases where you have some simple state a a handful of actions that can modify the state.